R/epiweek.R

```diff
@@ -58,17 +58,82 @@ epiweek_end <- function(epiweek_standard) {
 }
 
 
+#' Get a date-to-epiweek function from an
+#' epiweek standard.
+#'
+#' Get the function that converts dates to epiweek numbers
+#' for a given epiweek standard.
+#'
+#' @param epiweek_standard One of `"MMWR"` or `"USA"`
+#' (USA MMWR epiweek, starts on Sunday) and `"ISO"` (ISO
+#' week, starts on Monday). Not case-sensitive.
+#' Must be a single value.
+#' @return A function, either [lubridate::epiweek()] or
+#' [lubridate::isoweek()], depending on the value of
+#' `epiweek_standard`
+epiweek_fn <- function(epiweek_standard) {
+  checkmate::assert_scalar(
+    epiweek_standard
+  )
+  epiweek_standard <- stringr::str_to_lower(epiweek_standard)
+  epiweek_fns <- c(
+    "mmwr" = lubridate::epiweek,
+    "usa" = lubridate::epiweek,
+    "iso" = lubridate::isoweek
+  )
+
+  checkmate::assert_names(
+    epiweek_standard,
+    subset.of = names(epiweek_fns),
+  )
```

Comment:
this is fine, but `rlang::arg_match` or `base::match.arg()` would be more typical, I think.